pypeit/core/flexure.py

```diff
@@ -697,3 +704,507 @@ def calculate_image_offset(image, im_ref, nfit=3):
     # Return the RA and DEC shift, in pixels
     return popt[1] - ccorr.shape[0]//2, popt[2] - ccorr.shape[1]//2
 
+
+
+def sky_em_residuals(wave:np.ndarray, flux:np.ndarray,
+                     ivar:np.ndarray, sky_waves:np.ndarray,
+                     plot=False, noff=5., nfit_min=20):
+    """Calculate residuals and other metrics for a set of
+    auto-magically identified sky emission lines
+
+    Args:
+        wave (np.ndarray): Wavelengths (in air!)
+        flux (np.ndarray): 
+        ivar (np.ndarray): 
+        sky_waves (np.ndarray): Skyline wavelengths (in air!)
+        plot (bool, optional): Defaults to False.
+        noff (int, optional): Range in Ang to analyze labout emission line. Defaults to 5.
+        nfit_min (int, optional): Minimum number of pixels required to do a fit. Defaults to 20.
+
+    Returns:
+        tuple: np.ndarray's -- sky line wavelength, wavelength offset, 
+            error in wavelength offset, sky line width,
+            error in sky line width
+    """
+
+
+    dwave = []
+    diff = []
+    diff_err  = []
+    los = []
+    los_err= []
+    # Loop on known sky lines
+    for line in sky_waves: 
+        wline = [line-noff,line+noff] 
+        mw    = (wave > wline[0]) & (wave < wline[1])
+        
+        # Reuire minimum number
+        if np.sum(mw) <= nfit_min:
+            continue
+
+        p=[0,0,0,0]
+        # Guess
+        p0 = list(fitting.guess_gauss(wave[mw],flux[mw]))
+        # Fit
+        try:
+            p, pcov = fitting.fit_gauss(wave[mw],flux[mw], 
+                                w_out = 1./np.sqrt(ivar[mw]), 
+                                guesses=p0, nparam=4)
+            perr = np.sqrt(np.diag(pcov))
+        except:
```

Comment:
I fear I don't know what that tends to be.

Reply:
If you don't understand the failure mode, then there should not be a try except. In principle we could add such logic every time we do a fit "just to be safe", but this would not be a very robust way to code.

Reply:
ok, removed.  Have kept the lines around for future use, as needed.